src/sfctl/auth.py

```diff
@@ -32,3 +32,24 @@ def signed_session(self):
             session.verify = False
 
         return session
+
+class AdalAuthentication(Authentication):
+    """Azure Active Directory authentication for Service Fabric clusters"""
+    access_token = None
+
+    def __init__(self, token, no_verify=False):
+        self.access_token = token
+        self.no_verify = no_verify
+
+    def signed_session(self):
```

Comment:
I believe you also need to support refreshing the token if it expires, take a look at [this page on OAuth](http://azure-sdk-for-python.readthedocs.io/projects/msrest/en/latest/msrest.html#msrest.authentication.OAuthTokenAuthentication.refresh_session)